model/file_info.go

```diff
@@ -131,25 +128,25 @@ func GetInfoForBytes(name string, data []byte) (*FileInfo, *AppError) {
 
 	if info.IsImage() {
 		// Only set the width and height if it's actually an image that we can understand
-		if config, _, err := image.DecodeConfig(bytes.NewReader(data)); err == nil {
+		if config, _, err := image.DecodeConfig(file); err == nil {
 			info.Width = config.Width
 			info.Height = config.Height
-
 			if info.MimeType == "image/gif" {
 				// Just show the gif itself instead of a preview image for animated gifs
-				if gifConfig, err := gif.DecodeAll(bytes.NewReader(data)); err != nil {
+				if gifConfig, err := gif.DecodeAll(file); err != nil {
```

Comment:
I believe this is broken: an `io.Reader` can only be read once, and has no way to "rewind" the stream. The reader will have been consumed via the call to `image.DecodeConfig` on line 131, after which `gif.DecodeAll` will see an empty stream.